packages/reports/src/Stores/Modules/Profit/profit-store.js

```diff
@@ -135,8 +135,8 @@ export default class ProfitTableStore extends BaseStore {
         If we already have a cache for a period and if we sell a contract that was purchased in that period
         then the sold contract won't be there in profit_table when visited again unless we fetch it again.
         Caching will only work if the date filtering happens based on `sell_time` of a contract in BE. */
-        this.clearTable();
-        this.fetchNextBatch(shouldFilterContractTypes);
+        if (loginid) this.clearTable();
+        this.fetchNextBatch(shouldFilterContractTypes, !loginid);
```

Comment:
```suggestion
        if (loginid) { 
           this.clearTable();
           this.fetchNextBatch(shouldFilterContractTypes);
        }
```
 Will this work? We shouldn't be calling the `fetchNextBatch` if the loginid is not available.

Reply:
@balakrishna-deriv no worries, at this point, loginid is always available as `this.client_loginid` at line 129 above :) 
And here I'm using the cached value (saved before `this.client_loginid` had been set), so that I can know if it's the initial load.

Reply:
i'll remove this check for loginid since this onMount is only called when mounting